core/kernel.rbs

```diff
@@ -1547,7 +1577,16 @@ module Kernel : BasicObject
   #     (snipped)
   #     ping
   #
-  def self?.select: (::Array[IO] read, ?::Array[IO] write, ?::Array[IO] error, ?Integer timeout) -> ::Array[String]
+  def self?.select: (nil, nil, nil, _Timeout timeout) -> nil
+                  | [T < _ToIO] (Array[T]? read, ?Array[T]? write, ?Array[T]? errors, ?_Timeout? timeout) -> [Array[T], Array[T], Array[T]]
+
+  interface _Timeout
+    def divmod: (1) -> [int, _TimeoutNSecs]
+  end
+
+  interface _TimeoutNSecs
+    def *: (1000000000) -> int
+  end
```

Comment:
Implementation details talk aside – what classes beside Integer sastisfies the `_Timeout` interface, interstingly?
```ruby
4.5.divmod(1)[1] * 1000000000 #=> 500000000.0
```